Dependencies 추가

raindropiOS opened the pull request:
다음의 패키지를 CocoaPod으로 설치했습니다.
* SwiftLint
* Toaster

## SwiftLint
force_cast 규칙을 비활성화 했습니다.
현재 뷰 컨트롤러에서 타입캐스팅이 필요했습니다.  
```swift
let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "productCell",
                                                  for: indexPath) as! ProductCollectionViewCell
```

raindropiOS committed:
Add SwiftLint

raindropiOS committed:
Add SwiftLint run script

raindropiOS committed:
Add swiftlint_yml

raindropiOS committed:
Add toaster

godrm commented:
확인했습니다. 
pods install로 생기는 pods 폴더는 commit 에서 제외하셔야 합니다.
ignore에 넣고, git에서 remove하시면 될 겁니다.
이 부분 적용하고 머지하셔도 될 것 같네요

godrm commented:
> 



> force_cast 규칙을 비활성화 했습니다.
> 현재 뷰 컨트롤러에서 타입캐스팅이 필요했습니다.

타입 캐스팅을 하더라도 강제 캐스팅이 아니라 안전하게 처리하라는 의미일 겁니다
허용되는 문법도 확인해보세요

raindropiOS committed:
Untrack Pods directory

- gitignore에 Pods 경로를 추가했습니다.

hyeffie committed:
Uncheck "Based on dependency analysis" in the script phase

- 매 빌드마다 확인하도록 스크립트 옵션 변경

hyeffie committed:
Update Podfile.lock

- CocoaPods 업데이트 및 체크섬 업데이트
- Podfile 내용 수정

hyeffie committed:
Changed lint rules and apple new rules to the code.

- trailing_whitespace, trailing_comma 규칙 추가
- line length 규칙 120 -> 130
- force cast 규칙에 맞춰 as! 강제 캐스팅 연산자를 옵셔널 캐스팅으로 바꾸고 구현 수정
- 불필요한 line 제거
- supporting files 파일에 코딩 스타일 적용

hyeffie committed:
Remove prototype header from collection view

- 불필요한 프로토타입 header cell 제거

hyeffie commented:
원래 구현의 강제 캐스팅이 의도되었던 건데 코멘트를 보고 안전하게 처러하도록 구현을 바꾸기로 결정했습니다. lint 관련 옵션을 확인하고, 빌드 후 warnging 들을 확인하면서 disable할 default rule를 추가했고, 결정된 rule에 맞추어 코드를 좀 더 정리했습니다. pod 관련 내용도 추가 업데이트가 좀 있습니다. 리뷰 감사합니다!

raindropiOS committed:
Fix build phases order

- 컴파일 이전에 lint가 실행되도록 build phases 순서를 변경했습니다.
